tests/integration/test_api.py

```diff
@@ -470,3 +477,45 @@ def test_annotate_insight_anonymous_then_authenticated(client, mocker):
         server_domain=settings.OFF_SERVER_DOMAIN,
         auth=OFFAuthentication(username="a", password="b"),
     )
+
+
+def test_get_images(client):
+    result = client.simulate_get("/api/v1/images?page=1&count=25")
+    assert result.status_code == 200
+
+    image_model_factory1 = ImageModelFactory(barcode="123")
+    image_model_factory2 = ImageModelFactory(barcode="456")
+
+    image_prediction_factory1 = ImagePredictionFactory(image=image_model_factory1.id)
+    image_prediction_factory2 = ImagePredictionFactory(image=image_model_factory2.id)
+    image_prediction_factory3 = ImagePredictionFactory()
+
+    # test with "barcode" filter
+    image_model_data = get_images(barcode="123")
+    image_model_items = [item.to_dict() for item in image_model_data]
+    assert len(image_model_items) == 1
+    assert image_model_items[0]["id"] == image_model_factory1.id
+    assert image_model_items[0]["barcode"] == "123"
+
+    # test if "image_prediction" foreign key matches "image_model"
+    assert image_prediction_factory1.image.id == image_model_factory1.id
+    assert image_prediction_factory2.image.id == image_model_factory2.id
+
+    # test if foreign key does not match as not parameter is passed to image_prediction_factory3
+    assert not (image_prediction_factory3.image.id == image_model_factory1.id)
+    assert not (image_prediction_factory3.image.id == image_model_factory2.id)
+
+    # test filter with "barcode" and "with_predicted=False"
+    image_prediction_data = get_images(barcode="123", with_predicted=False)
+    image_prediction_items = [item.to_dict() for item in image_prediction_data]
+    assert len(image_prediction_items) == 1
```

Comment:
you already did this test above. instead test with `with_predicted=True` you should get two results this time.